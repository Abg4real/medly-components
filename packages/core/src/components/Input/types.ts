import { HTMLProps, WithThemeProp } from '@core-utils/types';

export interface Props extends HTMLProps<HTMLInputElement>, WithThemeProp {}
