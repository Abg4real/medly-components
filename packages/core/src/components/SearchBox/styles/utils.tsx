import { SvgIcon } from '@medly-components/icons';
import { css } from '@medly-components/utils';
import { Props } from '../types';

export const getSearchBoxSize = ({ theme, size }: Props) => theme.searchBox.sizes[size].height;

export const getBorderAndBoxShadow = (color: string, boxShadow: string) => {
    return css`
        border-color: ${color};
        box-shadow: ${boxShadow};
    `;
};

export const getIconWrapperStyle = ({ theme, size }: Props) => {
    return css`
        width: ${getSearchBoxSize};
        min-width: ${getSearchBoxSize};
        height: auto;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        ${SvgIcon} {
            cursor: pointer;
            font-size: ${theme.searchBox.sizes[size].iconSize};
        }
    `;
};
