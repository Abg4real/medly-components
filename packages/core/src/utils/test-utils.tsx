import { defaultTheme } from '@medly-components/theme';
import React from 'react';
import { render, RenderResult } from 'react-testing-library';
import { ThemeProvider } from './styled';

const AllTheProviders: React.FunctionComponent = props => (
    <ThemeProvider theme={defaultTheme}>
        <>{props.children}</>
    </ThemeProvider>
);

const customRender = (ui: React.ReactElement<any>, options?: object): RenderResult => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
