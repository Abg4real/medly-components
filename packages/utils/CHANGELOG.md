# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-beta.16](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.15...@medly-components/utils@1.0.0-beta.16) (2020-04-11)


### Features

* add typography and refactor components based on new typography ([462f2aa](https://github.com/medlypharmacy/medly-components/commit/462f2aa5b24c968b941887e6fcea5fd4e5c7b3f1))





# [1.0.0-beta.15](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.14...@medly-components/utils@1.0.0-beta.15) (2020-04-02)


### Bug Fixes

* **all:** fix lint issues ([ca12080](https://github.com/medlypharmacy/medly-components/commit/ca1208092c7a4d0ce0e22f20cb31c0400c4ddbff))





# [1.0.0-beta.14](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.13...@medly-components/utils@1.0.0-beta.14) (2020-03-30)


### Features

* **all:** add rollup to create bundles ([08b5972](https://github.com/medlypharmacy/medly-components/commit/08b5972f4a221037d14449ffce508f8656d0f447))





# [1.0.0-beta.13](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.12...@medly-components/utils@1.0.0-beta.13) (2020-03-29)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.12](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.11...@medly-components/utils@1.0.0-beta.12) (2020-02-06)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.11](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.10...@medly-components/utils@1.0.0-beta.11) (2020-02-04)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.10](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.9...@medly-components/utils@1.0.0-beta.10) (2020-01-22)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.9](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.8...@medly-components/utils@1.0.0-beta.9) (2020-01-21)


### Features

* **core:** add functionaity to accept date as string also in datepicker component ([3c3c377](https://github.com/medlypharmacy/medly-components/commit/3c3c377a66c470e6765d4a98f0f55b443b5ab0bc))
* **icons:** add bill icon ([2b461c2](https://github.com/medlypharmacy/medly-components/commit/2b461c2478590247498b08f13cad83d225efff6e))





# [1.0.0-beta.8](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.7...@medly-components/utils@1.0.0-beta.8) (2020-01-12)


### Features

* **all:** add side-effects to package.json ([0f78b4c](https://github.com/medlypharmacy/medly-components/commit/0f78b4c2ba432373b9d66861c25a265dacd5d433))





# [1.0.0-beta.7](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.6...@medly-components/utils@1.0.0-beta.7) (2020-01-09)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.6](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.5...@medly-components/utils@1.0.0-beta.6) (2019-12-17)


### Bug Fixes

* **core:** fix modal close issue ([f2d8fe9](https://github.com/medlypharmacy/medly-components/commit/f2d8fe9553f9efdbeb96c8b5b213844f0c472777))





# [1.0.0-beta.5](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.4...@medly-components/utils@1.0.0-beta.5) (2019-12-10)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.4](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.3...@medly-components/utils@1.0.0-beta.4) (2019-11-28)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.3](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.2...@medly-components/utils@1.0.0-beta.3) (2019-11-13)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.2](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.1...@medly-components/utils@1.0.0-beta.2) (2019-11-13)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.1](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.2.0...@medly-components/utils@1.0.0-beta.1) (2019-11-13)


### Bug Fixes

* **all:** remove changelog for pre release setup ([0795f5c](https://github.com/medlypharmacy/medly-components/commit/0795f5c139b529c3f6b65442ff71a8826ecf2c56))
* fixed various issues in date picker ([00fed84](https://github.com/medlypharmacy/medly-components/commit/00fed842b6d363f817de2c8703785c62557cf50d))


### Features

* **all:** remove s3 upload command from all packages ([89ea085](https://github.com/medlypharmacy/medly-components/commit/89ea085082829cba2098342c9793a43e2bc2e24a))
* **core:** add capability to change option in Single select through arrow keys ([fd864b9](https://github.com/medlypharmacy/medly-components/commit/fd864b9eb103bc042086c51236f10222cfc57367))
* added capitalize function to capitalize first letter of the string ([2613604](https://github.com/medlypharmacy/medly-components/commit/261360434dcd02de4cde438350d00696323d16d5))
* added DatePicker and DateRangePicker component ([5dc6d38](https://github.com/medlypharmacy/medly-components/commit/5dc6d38f66f6d1d530f830da13a9624715f6f704))
