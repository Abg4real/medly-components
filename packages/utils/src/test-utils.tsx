import { defaultTheme } from '@medly-components/theme';
import { render, RenderResult } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from './styled';

const AllTheProviders: React.FunctionComponent = props => (
    <ThemeProvider theme={defaultTheme}>
        <>{props.children}</>
    </ThemeProvider>
);

const customRender = (ui: React.ReactElement<any>, options?: object): RenderResult => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
