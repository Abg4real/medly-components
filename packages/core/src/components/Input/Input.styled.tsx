import { defaultTheme } from '@medly-components/theme';
import { css, fullWidth, oppositePositionalspacing, styled } from '@medly-components/utils';
import Text from '../Text';
import { Props } from './types';

export const InputStyled = styled('input').attrs(({ theme: { input, text, font } }) => ({ input, text, font }))<Props>`
    padding: ${({ theme }) => theme.spacing.S};
    margin: ${({ theme, labelPosition }) => oppositePositionalspacing(labelPosition, theme.spacing.S)};
    font-size: ${({ font, text }) => font.sizes[text.defaults.textSize]};
    font-weight: ${({ font, text }) => font.weights[text.defaults.textWeight]};
    color: ${props => props.input.color};
    background-color: ${props => props.input.bgColor};
    background-clip: padding-box;
    box-sizing: border-box;
    border: 1px solid ${props => props.input.borderColor};
    border-radius: 5px;

    &:focus {
        border-color: ${props => props.input.focus.borderColor};
        outline: 0;
    }

    &::placeholder {
        color: ${props => props.input.placeHolderColor};
    }

    ::-webkit-file-upload-button {
        visibility: hidden;
    }

    ${props => props.fullWidth && fullWidth()};

    ${({ description, theme }) =>
        description &&
        css`
            & ~ ${Text.Style} {
                color: ${theme.input.desciptionColor};
            }
        `}
`;

InputStyled.defaultProps = {
    theme: defaultTheme
};
