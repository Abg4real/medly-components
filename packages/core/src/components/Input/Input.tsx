import { defaultTheme } from '@medly-components/theme';
import { WithStyle, withTheme } from '@medly-components/utils';
import React from 'react';
import FieldWithLabel from '../FieldWithLabel';
import Label from '../Label';
import Text from '../Text';
import { InputStyled } from './Input.styled';
import { Props } from './types';

const Input: React.FunctionComponent<Props> & WithStyle = React.memo(
    React.forwardRef((props, ref) => {
        const { id, description, label, labelPosition, fullWidth, required } = props;
        return (
            <FieldWithLabel {...{ fullWidth, labelPosition }}>
                {label && (
                    <FieldWithLabel.Label htmlFor={id || 'input'} {...{ required, labelPosition }}>
                        {label}
                    </FieldWithLabel.Label>
                )}
                <InputStyled id={id || 'input'} ref={ref} {...props} />
                {description && <FieldWithLabel.Description>{description}</FieldWithLabel.Description>}
            </FieldWithLabel>
        );
    })
);

Input.displayName = 'Input';
Input.Style = InputStyled;
Input.defaultProps = {
    disabled: false,
    fullWidth: false,
    required: false,
    label: '',
    labelPosition: 'top',
    description: ''
};

export default Input;
