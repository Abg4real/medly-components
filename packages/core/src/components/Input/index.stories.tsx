import { wInfo } from '@storybook-utils';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Input from './Input';

storiesOf('Components', module).add(
    'Input',
    () => <Input placeholder="Enter Your Name" />,
    wInfo(`Input Component with all the default props`)
);
