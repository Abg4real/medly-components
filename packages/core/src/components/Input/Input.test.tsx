import { render } from '../../utils/test-utils';
import 'jest-styled-components';
import React from 'react';
import Input from './Input';

describe('Input component', () => {
    it('should render correctly', () => {
        const { container } = render(<Input type="email" />);
        expect(container).toMatchSnapshot();
    });
});
