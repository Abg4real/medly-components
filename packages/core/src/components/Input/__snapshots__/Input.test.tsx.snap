// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly when fullWidth & required props are falsy 1`] = `
.c2 {
  font-size: 14px;
  font-weight: 400;
  color: #555;
  margin: 0px 8px 0px 0px;
}

.c5 {
  font-size: 14px;
  font-weight: 400;
  color: inherit;
}

.c1 {
  grid-area: label;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4 {
  grid-area: description;
  margin: 5px 0px 0px 2px;
  color: #818A91;
}

.c0 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-columns: minmax(max-content,25%) minmax(max-content,80%);
  grid-template-areas: 'label field' 'random description';
}

.c0 .c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  padding: 8px;
  font-size: 14px;
  font-weight: 400;
  color: #555;
  background-color: #ffffff;
  background-clip: padding-box;
  box-sizing: border-box;
  border: 1px solid #CCCCCC;
  border-radius: 5px;
}

.c3:focus {
  border-color: #012040;
  outline: 0;
}

.c3::-webkit-input-placeholder {
  color: #999999;
}

.c3::-moz-placeholder {
  color: #999999;
}

.c3:-ms-input-placeholder {
  color: #999999;
}

.c3::placeholder {
  color: #999999;
}

.c3::-webkit-file-upload-button {
  visibility: hidden;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1 c2"
      for="email-input"
    >
      Email Address
    </label>
    <input
      class="c3"
      id="email-input"
      label="Email Address"
      placeholder="Enter Email Address"
      type="email"
    />
    <span
      class="c4 c5"
    >
      We will never share your email with anyone
    </span>
  </div>
</div>
`;

exports[`Input component should render correctly when it is disabled 1`] = `
.c0 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-areas: 'label' 'field' 'description';
  grid-template-columns: auto;
  grid-template-areas: 'field' 'description';
}

.c0 .c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  padding: 8px;
  font-size: 14px;
  font-weight: 400;
  color: #555;
  background-color: #ededed;
  background-clip: padding-box;
  box-sizing: border-box;
  border: 1px solid #CCCCCC;
  border-radius: 5px;
}

.c1:focus {
  border-color: #012040;
  outline: 0;
}

.c1::-webkit-input-placeholder {
  color: #999999;
}

.c1::-moz-placeholder {
  color: #999999;
}

.c1:-ms-input-placeholder {
  color: #999999;
}

.c1::placeholder {
  color: #999999;
}

.c1::-webkit-file-upload-button {
  visibility: hidden;
}

<div>
  <div
    class="c0"
  >
    
    <input
      class="c1"
      disabled=""
      id="email-input"
      label=""
      placeholder="Enter Email Address"
    />
    
  </div>
</div>
`;

exports[`Input component should render correctly when label and description are not given 1`] = `
.c0 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-areas: 'label' 'field' 'description';
  grid-template-columns: auto;
  grid-template-areas: 'field' 'description';
}

.c0 .c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  padding: 8px;
  font-size: 14px;
  font-weight: 400;
  color: #555;
  background-color: #ffffff;
  background-clip: padding-box;
  box-sizing: border-box;
  border: 1px solid #CCCCCC;
  border-radius: 5px;
}

.c1:focus {
  border-color: #012040;
  outline: 0;
}

.c1::-webkit-input-placeholder {
  color: #999999;
}

.c1::-moz-placeholder {
  color: #999999;
}

.c1:-ms-input-placeholder {
  color: #999999;
}

.c1::placeholder {
  color: #999999;
}

.c1::-webkit-file-upload-button {
  visibility: hidden;
}

<div>
  <div
    class="c0"
  >
    
    <input
      class="c1"
      id="email-input"
      label=""
      placeholder="Enter Email Address"
      type="email"
    />
    
  </div>
</div>
`;

exports[`Input component should render correctly with all the props given 1`] = `
.c2 {
  font-size: 14px;
  font-weight: 400;
  color: #555;
  margin: 0px 0px 8px 0px;
}

.c2::after {
  color: red;
  content: ' *';
}

.c5 {
  font-size: 14px;
  font-weight: 400;
  color: inherit;
}

.c1 {
  grid-area: label;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4 {
  grid-area: description;
  margin: 5px 0px 0px 2px;
  color: #818A91;
}

.c0 {
  margin: 8px 0px;
  display: grid;
  grid-template-areas: 'label' 'field' 'description';
}

.c0 .c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  padding: 8px;
  font-size: 14px;
  font-weight: 400;
  color: #555;
  background-color: #ffffff;
  background-clip: padding-box;
  box-sizing: border-box;
  border: 1px solid #CCCCCC;
  border-radius: 5px;
  width: calc(100% - 0px);
}

.c3:focus {
  border-color: #012040;
  outline: 0;
}

.c3::-webkit-input-placeholder {
  color: #999999;
}

.c3::-moz-placeholder {
  color: #999999;
}

.c3:-ms-input-placeholder {
  color: #999999;
}

.c3::placeholder {
  color: #999999;
}

.c3::-webkit-file-upload-button {
  visibility: hidden;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1 c2"
      for="email-input"
      required=""
    >
      Email Address
    </label>
    <input
      class="c3"
      id="email-input"
      label="Email Address"
      placeholder="Enter Email Address"
      required=""
      type="email"
    />
    <span
      class="c4 c5"
    >
      We will never share your email with anyone
    </span>
  </div>
</div>
`;
